### DISCUSSION TITLE
Could we add an option on videostreamplayer to make it can pause video when the video finished?

### DISCUSSION THREAD

**Discussion by Sasinim5** [godotengine/godot-proposals#10420] [Engine Core]
Since Godot script can not get the video length. If the videostreamplayer can pause the video when finished, we can do anything with video since it paused. at the same time, we know the video length via update steamposition in process thread.

**Comment by Calinou** [comment]
Doesn't video playback already pause when finished unless you enabled **Loop**? Or do you want the last frame to remain visible instead of a black screen?

I suppose you could work this around by taking a screenshot of the last video frame, adding the image to your project, and putting a TextureRect with this texture as the parent node of your VideoStreamPlayer (so it'll show once playback is done).

**Reply by Sasinim5** [reply]
When the video finished, i did not see the black screen. the video player looks like closed. I will try the way you said.

**Reply by Sasinim5** [reply]
By the other hand, the way of taking screenshots is a hacking way i think. I am afraid it may has transistion issue when swtich video to texture.

**Reply by Calinou** [reply]
> I am afraid it may has transistion issue when swtich video to texture.

The transition should be unnoticeable if the texture has the same resolution as the video.

**Reply by Sasinim5** [reply]
So, Do this thing in videostreamplayer finish video signal funciton?

**Reply by Sasinim5** [reply]
I tried, failed. The videostreamplayer closed once it finished. the last frame image is empty.

**Reply by etherealxx** [reply]
@Sasinim5 here's a better workaround i found

connect the `finished` signal of the videostreamplayer to this function
```gdscript
# where "video" is your videostreamplayer node
func _on_video_finished():
	video.play()
	video.paused = true
```

though honestly i wish there's a better in-engine implementation of this

**Reply by berarma** [reply]
> @Sasinim5 here's a better workaround i found
> 
> connect the `finished` signal of the videostreamplayer to this function
> 
> ```gdscript
> # where "video" is your videostreamplayer node
> func _on_video_finished():
> 	video.play()
> 	video.paused = true
> ```
> 
> though honestly i wish there's a better in-engine implementation of this

Currently, there's no need to do this. When the video ends the last frame stays. With Godot ~~4.4+~~ 4.5 at least.

**Reply by SteveSmith16384** [reply]
> Currently, there's no need to do this. When the video ends the last frame stays. With Godot 4.4+ at least.

I'm using 4.4.1.stable, and when my videos finish, they disappear.  The above code is needed to prevent this.

**Reply by berarma** [reply]
> > Currently, there's no need to do this. When the video ends the last frame stays. With Godot 4.4+ at least.
> 
> I'm using 4.4.1.stable, and when my videos finish, they disappear. The above code is needed to prevent this.

Sorry, I was wrong. It's solved in the current development branch (4.5). This change didn't make it in 4.4.x.
